assets/app/scripts/directives/editRequestLimit.js

```diff
@@ -0,0 +1,234 @@
+          // Set unit default based on default value or millicores/MiB if no default value.
+          // The amount input will have a placeholder using on limit range default.
+          var setDefault = _.spread(function(defaultAmount, defaultUnit) {
+            scope.placeholder = defaultAmount;
+            addUnitOption(defaultUnit);
+            // Only change selected unit if no value is set.
+            if (!scope.amount) {
+              scope.unit = defaultUnit;
+            }
+          });
+          console.log(defaultValue);
+          if (defaultValue) {
+            setDefault(amountAndUnit(defaultValue, scope.type));
+          }
+        });
+
+        // Set unit options and based on type.
+        switch (scope.type) {
+        case 'cpu':
+          scope.unit = 'm';
+          scope.units = [{
+            value: "m",
+            label: "millicores"
+          }, {
+            value: "",
+            label: "cores"
+          }];
+          break;
+        case 'memory':
+          scope.unit = 'Mi';
+          scope.units = [{
+            value: "M",
+            label: "MB"
+          }, {
+            value: "G",
+            label: "GB"
+          }, {
+            value: "Mi",
+            label: "MiB"
+          }, {
+            value: "Gi",
+            label: "GiB"
+          }];
+          break;
+        }
+
+        var validateLimitRange = function() {
+          // Use usageValue filter to normalize units for comparison.
+          var value = scope.amount && usageValue(scope.amount + scope.unit),
+              min = scope.limitRangeMin && usageValue(scope.limitRangeMin),
+              max = scope.limitRangeMax && usageValue(scope.limitRangeMax),
+              minValid = true,
+              maxValid = true;
+
+          // Test against limit range min if defined.
+          if (value && min) {
+            minValid = value >= min;
+          }
+
+          // Test against limit range max if defined.
+          if (value && max) {
+            maxValid = value <= max;
+          }
+
+          scope.form.amount.$setValidity('limitRangeMin', minValid);
+          scope.form.amount.$setValidity('limitRangeMax', maxValid);
+        };
+
+        var validateLimitAgainstRequest = function() {
+          // Use usageValue filter to normalize units for comparison.
+          var limit = scope.amount && usageValue(scope.amount + scope.unit),
+              request = scope.request && usageValue(scope.request),
+              limitLargerThanRequest = true,
+              limitWithinRatio = true;
+
+          if (request && limit) {
+            // Limit must be greater than or equal to request.
+            limitLargerThanRequest = limit >= request;
+
+            // Limit must be within the max limit/request ratio if defined.
+            if (scope.maxLimitRequestRatio) {
+              limitWithinRatio = (limit / request) <= scope.maxLimitRequestRatio;
```

Comment:
when request is set and limit isnt then you have an unbounded limit, if a a maxLimitRequestRatio is set then this pod would be rejected ( i just double checked with @derekwaynecarr )